lml/src/main/java/com/github/czyzby/lml/parser/impl/attribute/OnDoubleClickLmlAttribute.java

```diff
@@ -0,0 +1,73 @@
+package com.github.czyzby.lml.parser.impl.attribute;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.scenes.scene2d.Actor;
+import com.badlogic.gdx.scenes.scene2d.InputEvent;
+import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
+import com.github.czyzby.lml.parser.LmlParser;
+import com.github.czyzby.lml.parser.action.ActorConsumer;
+import com.github.czyzby.lml.parser.tag.LmlAttribute;
+import com.github.czyzby.lml.parser.tag.LmlTag;
+
+/** Attaches a ClickListener to the the actor, invoking a chosen action upon left button double clicking on actor. Expects an action ID.
+ * Assigned action method might have {@link OnDoubleClickLmlAttribute.Params} as parameter to retrieve some event related data.
+ * Mapped to "onDoubleClick" and "doubleClick" attribute names.
+ *
+ * @author Metaphore */
+public class OnDoubleClickLmlAttribute implements LmlAttribute<Actor> {
+    private static final long SECOND_CLICK_TIME = 250000000L; // 250ms
+
+    @Override
+    public Class<Actor> getHandledType() {
+        return Actor.class;
+    }
+
+    @Override
+    public void process(final LmlParser parser, final LmlTag tag, final Actor actor, final String rawAttributeData) {
+        final ActorConsumer<?, Params> action = parser.parseAction(rawAttributeData, tmpParams);
+        if (action == null) {
+            parser.throwError("Could not find action for: " + rawAttributeData + " with actor: " + actor);
+        }
+        actor.addListener(new ClickListener(0) {
+            private boolean firstClickCaught = false;
+            private long lastClickTime = 0;
+
+            @Override
+            public void clicked(final InputEvent event, final float x, final float y) {
+                long currentEventTime = Gdx.input.getCurrentEventTime();
+                long deltaTime = currentEventTime - lastClickTime;
+                lastClickTime = currentEventTime;
+
+                if (!firstClickCaught) {
+                    firstClickCaught = true;
+                } else {
+                    if (deltaTime < SECOND_CLICK_TIME) {
+                        firstClickCaught = false;
+
+                        tmpParams.actor = actor;
+                        tmpParams.x = x;
+                        tmpParams.y = y;
+                        tmpParams.stageX = event.getStageX();
+                        tmpParams.stageY = event.getStageY();
+                        action.consume(tmpParams);
+                        tmpParams.reset();
+                    }
+                }
+            }
+        });
+    }
+
+    private static Params tmpParams = new Params();
```

Comment:
I'd keep a single cache per a listener or simply add the fields to the listener. You'd have to create _a lot_ of listeners for `Params` objects to become an issue - and this approach might introduce potential multi-threading issues.

Reply:
I see your point, will refactor to cache per instance. About multi-threading, what is a point of your concern, LibGDX calls all input listeners in a main thread, isn't it?

Reply:
It should, unless you manually invoke the listener - which you shouldn't do. I'm more concerned about the user keeping a reference to the `Params` object and treating it as if it was an immutable object. I just wouldn't use cache/pooling unless it really becomes an issue.